tests/test_filter_count.py

```diff
@@ -104,11 +105,108 @@ def make_data(
     }
 
 
+def make_multilabel_data(
+    means=[[-5, 2], [0, 2], [-3, 6]],
+    covs=[[[3, -1.5], [-1.5, 1]], [[1, 0.5], [0.5, 4]], [[5, 1], [1, 5]]],
+    means_mul=[[-3, 1.5], [-6, 4], [2, 6], [-2, 3]],
+    covs_mul=[
+        [[0.3, 0.0], [0.0, 0.4]],
+        [[0.3, 0.0], [0.0, 0.4]],
+        [[0.3, 0.0], [0.0, 0.4]],
+        [[0.3, 0.0], [0.0, 0.4]],
+        [[0.2, 0], [0, 0.2]],
+    ],
+    label_mul=[[0, 1], [0, 2], [1, 2], [0, 1, 2]],
+    sizes_single_label=[100, 150, 100],
+    sizes_multi_label=[10, 20, 20, 5],
+    avg_trace=0.9,
+    seed=1,
+):
+    m = len(means) + len(
+        means_mul
+    )  # number of classes by treating each multilabel as 1 unique label
+    n = sum(sizes_single_label) + sum(sizes_multi_label)
+    local_data = []
+    labels = []
+    test_data = []
+    test_labels = []
+    for i in range(0, len(means)):
+        local_data.append(
+            np.random.multivariate_normal(mean=means[i], cov=covs[i], size=sizes_single_label[i])
+        )
+        test_data.append(
+            np.random.multivariate_normal(mean=means[i], cov=covs[i], size=sizes_single_label[i])
+        )
+        test_labels += [[i]] * sizes_single_label[i]
+        labels += [[i]] * sizes_single_label[i]
+
+    for i in range(0, len(means_mul)):
+        local_data.append(
+            np.random.multivariate_normal(
+                mean=means_mul[i], cov=covs_mul[i], size=sizes_multi_label[i]
+            )
+        )
+        test_data.append(
+            np.random.multivariate_normal(
+                mean=means_mul[i], cov=covs_mul[i], size=sizes_multi_label[i]
+            )
+        )
+        test_labels += [label_mul[i]] * sizes_multi_label[i]
+        labels += [label_mul[i]] * sizes_multi_label[i]
+
+    X_train = np.vstack(local_data)
+    X_test = np.vstack(test_data)
+
+    d = {}
+    for i in labels:
+        if str(i) not in d:
+            d[str(i)] = len(d)
+    inv_d = {v: k for k, v in d.items()}
+
+    labels_idx = [d[str(i)] for i in labels]
+
+    py = np.bincount(labels_idx) / float(len(labels_idx))
+
+    noise_matrix = generate_noise_matrix_from_trace(
+        m,
+        trace=avg_trace * m,
+        py=py,
+        valid_noise_matrix=True,
+        seed=seed,
+    )
+    noisy_labels_idx = generate_noisy_labels(labels_idx, noise_matrix)
+    noisy_labels = [eval(inv_d[i]) for i in noisy_labels_idx]
+    ps = np.bincount(labels_idx) / float(len(labels_idx))
+    inv = compute_inv_noise_matrix(py, noise_matrix, ps=ps)
+
+    clf = MultiOutputClassifier(LogisticRegression())
+    y_train = int2onehot(noisy_labels)
+    clf.fit(X_train, y_train)
+    pyi = clf.predict_proba(X_train)
+    pred_probs = -1 * np.ones(y_train.shape)
```

Comment:
could you explain why it is initialized like this, sorry don't exactly follow it

Reply:
it's from PoHe's code, 
I can initialize this with zeros,

Reply:
oh ok so there is no reason to initialize it this way? It's just being overwritten right?

Reply:
yes it's being overridden, can be initialized with zeros, 
just faster than assigning a list and then converting to array